### ISSUE TITLE
Preserving comments

### ISSUE THREAD

**Issue by paulvi** [takari/polyglot-maven#85]
While translating (e.g. migrating from .xml) information written in comments is lost.

Comments before/after any section can't be in Model, so it is for translators to keep more data

**Comment by mosabua** [comment]
This seems to be a non-crucial feature and I am inclined to close the issue. Any objections? Or anybody wanting to send a PR that implements it for the various dialects? Or for one as an example at least?

**Comment by paulvi** [comment]
objection. I consider `pom` as code, and comments to be important part of it.

**Comment by mosabua** [comment]
Fair point. Wanna send a PR?

**Comment by mkristian** [comment]
we can do translate POMs back and forth. the translation from pom.xml to pom.rb is straight forward, the other way around you have builder methods here and there, so a lot of coding around the actual DSL. I doubt a general translation of comments to xml-comments is feasible and can be meaningful.

**Comment by lefou** [comment]
Although I'm not saying preserving the comments isn't possible (in fact, I have some ideas), I would argue, it is not worth the effort. Or at least not for those dialects representing full programming languages like the ruby or scala dialect.

In most cases I use the translation from XML only once, to have an initial starting point. Later I use the translation to XML whenever I need integration into other tools. But the output looks more like the result of `help:effective-pom`, which doesn't preserve comments neither (I didn't checked). As the generated XML has a very different struckture than the source `pom.scala`, any preserved comment would have a high potential to tell a lie.

**Comment by lefou** [comment]
What I really want is a single most prominent comment saying, that the file was generated. But that's another story.

**Comment by jdimeo** [comment]
I would vote more for preserving comments from original translation from XML as being more important than preserving them back to XML after editing
